src/extension/server/runmeServer.ts

```diff
@@ -0,0 +1,97 @@
+
+import { ChildProcessWithoutNullStreams, spawn } from 'node:child_process'
+import path from 'node:path'
+import fs from 'node:fs/promises'
+
+import { Disposable, workspace } from 'vscode'
+
+import { SERVER_ADDRESS } from '../../constants'
+
+import RunmeServerError from './runmeServerError'
+
+export interface IServerConfig {
+    assignPortDynamically?: boolean
+    retryOnFailure?: boolean
+    maxNumberOfIntents: number
+}
+
+class RunmeServer implements Disposable {
+
+    #runningPort: number
+    #process: ChildProcessWithoutNullStreams | undefined
+    #binaryPath: string
+    #retryOnFailure: boolean
+    #maxNumberOfIntents: number
+    #loggingEnabled: boolean
+    #intent: number
+    #address: string
+
+    constructor(options: IServerConfig) {
+        const config = workspace.getConfiguration('runme.server')
+        this.#runningPort = config.get<number>('port')!
+        this.#loggingEnabled = config.get<boolean>('enableLogger', false)
+        this.#binaryPath = path.join(__dirname, '../', config.get<string>('binaryPath')!)
+        this.#retryOnFailure = options.retryOnFailure || false
+        this.#maxNumberOfIntents = options.maxNumberOfIntents
+        this.#intent = 0
+        this.#address = `${SERVER_ADDRESS}:${this.#runningPort}`
```

Comment:
I'd make sure this is really an int here... cmd-line injections since it's set by the user.

Reply:
Moved to zod schema